src/Microsoft.OData.Core/ODataParameterReaderCore.cs

```diff
@@ -432,6 +434,27 @@ protected virtual Task<bool> ReadAsynchronously()
             return TaskUtils.GetTaskForSynchronousOperation<bool>(this.ReadImplementation);

```

Comment:
Shouldn't this be "truly async" as well?

Reply:
@habbes The "truly async" implementation of `ReadAsynchronously` is contained in `ODataParameterReaderCoreAsync`. Because `ReadAsync` method is declared as `abstract` and `ODataParameterReaderCore` needs to override it, we're just wrapping the synchronous method in a task as a stub implementation.

Reply:
Oh, is that worth mentioning in a comment?